examples/hmcecs.py

```diff
@@ -0,0 +1,147 @@
+    mcmc.run(mcmc_key, data, obs, None)
+    mcmc.print_summary()
+    return mcmc.get_samples()
+
+
+def main(args):
+    assert 11_000_000 >= args.num_datapoints, "11,000,000 data points in the Higgs dataset"
+    # full dataset takes hours for plain hmc!
+    _, fetch = load_dataset(HIGGS, shuffle=False, num_datapoints=args.num_datapoints)
+    data, obs = fetch()
+
+    hmcecs_key, hmc_key = random.split(random.PRNGKey(args.rng_seed))
+
+    # choose inner_kernel
+    if args.inner_kernel.lower() == 'hmc':
+        inner_kernel = HMC(model)
+    elif args.inner_kernel.lower() == 'nuts':
+        inner_kernel = NUTS(model)
+    else:
+        inner_kernel = None
+
+    start = time.time()
+    losses, hmcecs_samples = run_hmcecs(hmcecs_key, args, data, obs, inner_kernel)
+    hmcecs_runtime = time.time() - start
+
+    if inner_kernel:
+        start = time.time()
+        hmc_samples = run_hmc(hmc_key, args, data, obs, inner_kernel)
+        hmc_runtime = time.time() - start
+
+        summary_plot(losses, hmc_samples, hmcecs_samples, hmc_runtime, hmcecs_runtime)
+
+
+def summary_plot(losses, hmc_samples, hmcecs_samples, hmc_runtime, hmcecs_runtime):
+    fig, ax = plt.subplots(2, 2)
+    ax[0, 0].plot(losses, 'r')
+    ax[0, 0].set_title('SVI losses')
+    ax[0, 0].set_ylabel('ELBO')
+
+    if hmc_runtime > hmcecs_runtime:
+        ax[0, 1].bar([0], hmc_runtime, label='hmc', color='b')
+        ax[0, 1].bar([0], hmcecs_runtime, label='hmcecs', color='r')
+    else:
+        ax[0, 1].bar([0], hmcecs_runtime, label='hmcecs', color='r')
+        ax[0, 1].bar([0], hmc_runtime, label='hmc', color='b')
+    ax[0, 1].set_title('Runtime')
+    ax[0, 1].set_ylabel('Seconds')
+    ax[0, 1].legend()
+    ax[0, 1].set_xticks([])
+
+    ax[1, 0].plot(jnp.sort(hmc_samples['theta'].mean(0)), 'or')
+    ax[1, 0].plot(jnp.sort(hmcecs_samples['theta'].mean(0)), 'b')
+    ax[1, 0].set_title(r'$\mathrm{\mathbb{E}}[\theta]$')
+
+    ax[1, 1].plot(jnp.sort(hmc_samples['theta'].var(0)), 'or')
+    ax[1, 1].plot(jnp.sort(hmcecs_samples['theta'].var(0)), 'b')
+    ax[1, 1].set_title(r'Var$[\theta]$')
+
+    for a in ax[1, :]:
+        a.set_xticks([])
+
+    fig.tight_layout()
+    fig.savefig('hmcecs_plot.pdf', bbox_inches='tight', transparent=True)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser("Hamiltonian Monte Carlo with Energy Conserving Subsampling")
+    parser.add_argument('--subsample_size', type=int, default=1300)
+    parser.add_argument('--num_svi_steps', type=int, default=5000)
+    parser.add_argument('--num_blocks', type=int, default=100)
+    parser.add_argument('--num_warmup', type=int, default=500)
+    parser.add_argument('--num_samples', type=int, default=500)
+    parser.add_argument('--num_datapoints', type=int, default=1_500_000)
+    parser.add_argument('--inner_kernel', type=str, default='nuts')
+    parser.add_argument('--device', default='cpu', type=str, help='use "cpu" or "gpu".')
```

Comment:
nit: use `choices=['nuts', 'hmc']` instead of `lower` etc

Reply:
Make sense; I'll update it.